staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -1736,6 +1736,10 @@ type CSIPersistentVolumeSource struct {
 	// Ex. "ext4", "xfs", "ntfs". Implicitly inferred to be "ext4" if unspecified.
 	// +optional
 	FSType string `json:"fsType,omitempty" protobuf:"bytes,4,opt,name=fsType"`
+
+	// Attributes of the volume to publish.
+	// +optional
+	VolumeAttributes map[string]string `json:"volumeAttributes,omitempty" protobuf:"bytes,5,rep,name=volumeAttributes"`
```

Comment:
As Saad mentioned, don't forget to re-generate the API code for this type.  If you need help, let me know.

Reply:
@musse When you change the API object, you must re-generate API source code.  Here are some steps:

0) Make sure to make you API change in `pkg/apis/core/v1/types.go` also.
1) From the kubernetes root dir, run `make update`
2) Follow directions when that task fails, it should walk you through what needs to be done
2) Run `make verify` to ensure all is good.  Follow directions there when things are broken.

Post here (or ping me on slack) if things dont work.

Reply:
I don't have the `pkg/apis/core/v1/types.go` file. Will it be generate with `make update`?

Reply:
@musse Apologies, this is the file and line# https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/types.go#L1607